src/util/delay.test.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-env jest */
+
+import delay from './delay'
+jest.useFakeTimers()
+
+describe('delay', () => {
+    test('should return with the given delay', () => {
+        delay(1000)
+        expect(setTimeout.mock.calls.length).toBe(1)
+        expect(setTimeout.mock.calls[0][1]).toBe(1000)
+    })
+
+    test('should return a promise which resolves', () => {
```

Comment:
Just move the `expect(assertionFunc).toBeCalled()` call to the end of the test. The rest should be ok. I didn't know that there is actually no public Promise API for checking if the promise is resolved or rejected.